src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -325,6 +329,53 @@ EnsureCitusTableCanBeCreated(Oid relationOid)
 }
 
 
+/*
+ * undistribute_table gets a distributed table name and 
+ * udistributes the table.
+ */
+Datum
+undistribute_table(PG_FUNCTION_ARGS)
+{
+	CheckCitusVersion(ERROR);
+	EnsureCoordinator();
+
+	Oid relationId = PG_GETARG_OID(0);
+	const char *relationName = get_rel_name(relationId);
+
+	EnsureTableOwner(relationId);
+
+	StringInfo query = makeStringInfo();
+
+	SPI_connect();
+
+	appendStringInfo(query, "CREATE TABLE tmp1 (LIKE %s INCLUDING ALL)", relationName);
```

Comment:
will the naming of indexes, constraints by identical to the original table?

Reply:
Probably not in an identical manner. In my opinion, `GetTableDDLEvents` might be used in this case. However, as I understood, you don't drop the distributed table for a moment. Hence directly executing the commands from `GetTableDDLEvents` wouldn't work. Maybe it would require some additional work after calling `ParseTreeCommand`

Reply:
Yes, it seems to just create them with new names. I'm looking into `GetTableDDLEvents`.